cobc/codegen.c

```diff
@@ -2763,8 +2762,28 @@ output_source_cache (void)
 
 /* Literal */
 
+static void
+cb_cache_program_decimal_constant (struct cb_program *prog, struct literal_list *cached_literal)
+{
+	struct literal_list	*l;
+	if (prog != NULL && cached_literal->make_decimal) {
+		for (l = prog->decimal_constants; l; l = l->next) {
+			if (cached_literal->id == l->id) {
+				return;
+			}
+		}
+
+		l = cobc_parse_malloc (sizeof (struct literal_list));
+		l->id = cached_literal->id;
+		l->literal = cached_literal->literal;
+		l->make_decimal = cached_literal->make_decimal;
+		l->next = prog->decimal_constants;
+		prog->decimal_constants = l;
+	}
+}
+
 int
-cb_lookup_literal (cb_tree x, int make_decimal)
+cb_lookup_literal (struct cb_program *prog, cb_tree x, int make_decimal)
```

Comment:
this one for example, could use one saying that the literal id is resolved from the literal cache and the literal added, if missing and after the change also say that if the literal is used as decimal, it is attached to the passed program's decimal constants list